uber/site_sections/preregistration.py

```diff
@@ -372,7 +398,11 @@ def prereg_payment(self, session, payment_id=None, stripeToken=None):
             message = 'Our preregistration price has gone up; ' \
                 'please fill out the payment form again at the higher price'
         else:
-            message = charge.charge_cc(session, stripeToken)
+            for attendee in charge.attendees:
+                message = check_prereg_promo_code(session, attendee)
```

Comment:
I think you should add an ``if not message:`` check so that we don't re-run the same check over and over again if it's already failed.